apigateway/internal/v1router/auth.go

```diff
@@ -1,50 +1,29 @@
 package v1router
 
 import (
-	"encoding/base64"
-	"math/rand"
-	"time"
-
 	"github.com/GrabItYourself/giys-backend/apigateway/internal/middlewares"
 	"github.com/GrabItYourself/giys-backend/apigateway/internal/v1router/types"
+	"github.com/GrabItYourself/giys-backend/auth/pkg/authproto"
 	"github.com/gofiber/fiber/v2"
 )
 
-const OAuthCookieName = "oauth-state"
-
-func generateOAuthStateCookie() (*fiber.Cookie, string) {
-	var expiration = time.Now().Add(365 * 24 * time.Hour)
-	b := make([]byte, 16)
-	rand.Read(b)
-	state := base64.URLEncoding.EncodeToString(b)
-
-	return &fiber.Cookie{
-		Name:     OAuthCookieName,
-		Value:    state,
-		Expires:  expiration,
-		HTTPOnly: true,
-	}, state
-}
-
 func (r *Router) InitAuthRoutes(basePath string) {
 	f := r.Router.Group(basePath)
 
-	f.Get("/google", func(c *fiber.Ctx) error {
-		cookie, state := generateOAuthStateCookie()
-		c.Cookie(cookie)
-		url := r.Handler.OAuthConfig.AuthCodeURL(state)
-		return c.Redirect(url)
-	})
-
-	f.Get("/google/callback", func(c *fiber.Ctx) error {
-		state := c.Query("state")
-		cookie := c.Cookies(OAuthCookieName)
-		if state != cookie {
-			return fiber.NewError(fiber.StatusBadRequest, "Invalid OAuth state")
-		}
-
+	f.Post("/google/verify", func(c *fiber.Ctx) error {
 		code := c.Query("code")
-		resp, err := r.Handler.HandleGoogleOAuthCallback(c.Context(), code)
+		if code == "" {
+			return fiber.NewError(fiber.StatusBadRequest, "code is required")
+		}
+		clientType := c.Query("clientType")
+		if clientType == "" {
+			return fiber.NewError(fiber.StatusBadRequest, "clientType is required")
+		}
```

Comment:
validate client type using this value https://api.flutter.dev/flutter/package-platform_platform/Platform/operatingSystemValues-constant.html